aws/rust-runtime/aws-types/src/config.rs

```diff
@@ -102,6 +111,105 @@ impl Builder {
         self
     }
 
+    /// Set the [`TimeoutConfig`] for the builder
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use std::time::Duration;
+    /// use aws_types::config::Config;
+    /// use aws_smithy_types::timeout::TimeoutConfig;
+    ///
+    /// let timeout_config = TimeoutConfig::new()
+    ///     .with_api_call_attempt_timeout(Some(Duration::from_secs(1)));
+    /// let config = Config::builder().timeout_config(timeout_config).build();
+    /// ```
+    pub fn timeout_config(mut self, timeout_config: TimeoutConfig) -> Self {
+        self.set_timeout_config(Some(timeout_config));
+        self
+    }
+
+    /// Set the [`TimeoutConfig`] for the builder
+    ///
+    /// # Examples
+    /// ```rust
+    /// # use std::time::Duration;
+    /// use aws_types::config::{Config, Builder};
+    /// use aws_smithy_types::timeout::TimeoutConfig;
+    ///
+    /// fn set_preferred_timeouts(builder: &mut Builder) {
+    ///     let timeout_config = TimeoutConfig::new()
+    ///         .with_api_call_attempt_timeout(Some(Duration::from_secs(2)))
+    ///         .with_api_call_timeout(Some(Duration::from_secs(5)));
+    ///     builder.set_timeout_config(Some(timeout_config));
+    /// }
+    ///
+    /// let mut builder = Config::builder();
+    /// set_preferred_timeouts(&mut builder);
+    /// let config = builder.build();
+    /// ```
+    pub fn set_timeout_config(&mut self, timeout_config: Option<TimeoutConfig>) -> &mut Self {
+        self.timeout_config = timeout_config;
+        self
+    }
+
+    /// Set the sleep implementation for the builder. The sleep implementation is used to create
+    /// timeout futures.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use std::sync::Arc;
+    /// use aws_smithy_async::rt::sleep::{AsyncSleep, Sleep};
+    /// use aws_types::config::Config;
+    ///
+    /// ##[derive(Debug)]
+    /// pub struct ForeverSleep;
+    ///
+    /// impl AsyncSleep for ForeverSleep {
+    ///     fn sleep(&self, duration: std::time::Duration) -> Sleep {
+    ///         Sleep::new(std::future::pending())
+    ///     }
+    /// }
+    ///
+    /// let sleep_impl = Arc::new(ForeverSleep);
+    /// let config = Config::builder().sleep_impl(sleep_impl).build();
+    /// ```
+    pub fn sleep_impl(mut self, sleep_impl: Arc<dyn AsyncSleep>) -> Self {
+        self.set_sleep_impl(Some(sleep_impl));
+        self
+    }
```

Comment:
can you make this doc hidden & the docs below for now? I think we will probably change the type to something else at some point